### PR TITLE
Fixed support for Node 8.12.0.

### PR DESCRIPTION
**Pull request by mcollina**

### CODE REVIEW

File: lib/v8-log-to-ticks.js [hunk 1]
```diff
@@ -22,8 +43,11 @@ function v8LogToTicks (isolateLogPath, node) {
     // it's an error in the V8 internal tick profiler)
     // ignore them.
     const ignore = /unknown code state:/
+    // pump(sp.stderr, process.stderr)
```
Comment (BridgeAR) [inline lib/v8-log-to-ticks.js]: Obsolete comment? :)

File: lib/v8-log-to-ticks.js [hunk 2]
```diff
@@ -3,11 +3,32 @@
...
 const fs = require('fs')
 const pump = require('pump')
 
-module.exports = v8LogToTicks
+const readFile = promisify(fs.readFile)
+const writeFile = promisify(fs.writeFile)
+
+module.exports = process
+
+function process (isolateLogPath, node) {
```
Comment (BridgeAR) [inline lib/v8-log-to-ticks.js]: Can we use a different name? Having the same name as global seems not ideal.

File: lib/v8-log-to-ticks.js [hunk 3]
```diff
@@ -3,11 +3,32 @@
...
+  return filterLinesThatAreTooLong(isolateLogPath).then((filtered) => {
```
Comment (BridgeAR) [inline lib/v8-log-to-ticks.js]: I suggest to only execute this when it's necessary (by checking the Node.js version).

File: lib/v8-log-to-ticks.js [hunk 4]
```diff
@@ -3,11 +3,32 @@
...
+
+// long lines make the --preprocess crash. We are just filtering them beforehand
+// as Node.js 8.12.0 would crash if there is any error in the preprocess script
+function filterLinesThatAreTooLong (isolateLogPath) {
+  const dest = isolateLogPath + '.filtered'
+  // TODO use streams
+  return readFile(isolateLogPath, 'utf8').then((data) => {
+    data = data.split('\n').filter((s) => s.length < 1024).join('\n')
+    return writeFile(dest, data)
+  }).then(() => dest)
```
Comment (BridgeAR) [inline lib/v8-log-to-ticks.js]: Suggestion: use async / await.

**Comment by mcollina** [comment]
PTAL

**Comment by mcollina** [comment]
Those lines cause the processor to fail, and we silence that error. I think this is just avoid future problems.

**Comment by zbjornson** [comment]
@mcollina this doesn't seem to always fix the problem -- I'm still getting the error with lines > 1024 characters removed. Maybe something else besides long lines causes `printErr` to be invoked?

**Comment by mcollina** [comment]
Can you upload an example to reproduce? Thanks!

Il giorno mer 17 ott 2018 alle 04:00 Zach Bjornson <notifications@github.com>
ha scritto:

**Pull request merged**